x-pack/legacy/plugins/apm/public/hooks/useTransactionBreakdown.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useRef } from 'react';
+import { useFetcher } from './useFetcher';
+import { callApi } from '../services/rest/callApi';
+import { getUiFiltersES } from '../services/ui_filters/get_ui_filters_es';
+import { TransactionBreakdownAPIResponse } from '../../server/lib/transactions/breakdown';
+import { useUrlParams } from './useUrlParams';
+
+export function useTransactionBreakdown() {
+  const {
+    urlParams: { serviceName, start, end, transactionName },
+    uiFilters
+  } = useUrlParams();
+
+  const { data, error, status } = useFetcher(
+    async () => {
+      if (serviceName && start && end) {
+        return callApi<TransactionBreakdownAPIResponse>({
```

Comment:
We have a convention of keeping rest calls separately in `services/rest`. Any reason to not do that here?

Reply:
I think we previously discussed this shortly, whether this abstraction (of placing rest calls in a separate file) was worth it. I opted to give it a shot without one. Will we use the call without a hook? If we do use a separate file, do we need a separate file with the hook as well?